src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -72,66 +106,69 @@ def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_c
+                                            membership_group_csv_urls, membership_group_dir_path)
 
-    export_traced_data(messages_by_column, f"{output_dir}/messages.jsonl")
-    export_traced_data(participants_by_column, f"{output_dir}/participants.jsonl")
+        export_analysis_file(messages_by_column, pipeline_config, f"{output_dir}/{channel}/messages.csv", export_timestamps=True)
+        export_analysis_file(participants_by_column, pipeline_config, f"{output_dir}/{channel}/participants.csv")
 
-    run_automated_analysis(messages_by_column, participants_by_column, pipeline_config.analysis, f"{output_dir}/automated-analysis")
+        export_traced_data(messages_by_column, f"{output_dir}/{channel}/messages.jsonl")
+        export_traced_data(participants_by_column, f"{output_dir}/{channel}/participants.jsonl")
 
-    dry_run_text = "(dry run)" if dry_run else ""
-    if pipeline_config.analysis.google_drive_upload is None:
-        log.debug(f"Not uploading to Google Drive, because the 'google_drive_upload' configuration was None {dry_run_text}")
-    else:
-        if dry_run:
-            log.info(f"Not uploading to Google Drive {dry_run_text}")
+        run_automated_analysis(messages_by_column, participants_by_column, pipeline_config.analysis, f"{output_dir}/{channel}/automated-analysis")
+
+        dry_run_text = "(dry run)" if dry_run else ""
+        if pipeline_config.analysis.google_drive_upload is None:
+            log.debug(f"Not uploading to Google Drive, because the 'google_drive_upload' configuration was None {dry_run_text}")
         else:
-            log.info("Uploading outputs to Google Drive...")
-            google_drive_upload.init_client(
-                google_cloud_credentials_file_path,
-                pipeline_config.analysis.google_drive_upload.credentials_file_url
-            )
-
-            drive_dir = pipeline_config.analysis.google_drive_upload.drive_dir
-            google_drive_upload.upload_file(f"{output_dir}/production.csv", drive_dir)
-            google_drive_upload.upload_file(f"{output_dir}/messages.csv", drive_dir)
-            google_drive_upload.upload_file(f"{output_dir}/participants.csv", drive_dir)
-            google_drive_upload.upload_all_files_in_dir(
-                f"{output_dir}/automated-analysis", f"{drive_dir}/automated-analysis", recursive=True
-            )
+            if dry_run:
+                log.info(f"Not uploading to Google Drive {dry_run_text}")
+            else:
+                log.info("Uploading outputs to Google Drive...")
+                google_drive_upload.init_client(
+                    google_cloud_credentials_file_path,
+                    pipeline_config.analysis.google_drive_upload.credentials_file_url
+                )
+
+                drive_dir = pipeline_config.analysis.google_drive_upload.drive_dir
+                google_drive_upload.upload_file(f"{output_dir}/{channel}/production.csv", drive_dir)
```

Comment:
Consider moving the analysis script for a group to its own function, so that it doesn't need to worry about channels, only the data and output_dir.

Reply:
Thanks, updated.